app/models/manageiq/providers/openstack/manager_mixin.rb

```diff
@@ -154,7 +154,9 @@ def translate_exception(err)
       when Excon::Errors::Timeout
         MiqException::MiqUnreachableError.new("Login attempt timed out")
       when Excon::Errors::SocketError
-        MiqException::MiqHostError.new("Socket error: #{err.message}")
+        MiqException::MiqHostError.new("Socket error: #{err.socket_error}")
+      when Excon::Error::BadRequest
+        MiqException::MiqHostError.new("Bad request: #{err.response.body}")
```

Comment:
NOTE: we could make the BadRequest error more readable if we parsed the XML but unsure if the response is _always_ XML or I need to handle the fallback case and just print the response body as-is, will probably go this route

Reply:
I think this is a good start, and we can get fancy later